src/modules/Game.class.js

```diff
@@ -1,68 +1,215 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
   constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.INITIAL_STATE = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.board = initialState || this.INITIAL_STATE;
+    this.score = 0;
+    this.status = 'idle';
+  }
+
+  moveLeft() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    this.board.forEach((row, rowIndex) => {
+      const originalRow = [...row];
+      const newRow = this.mergeRow(row);
+
+      this.board[rowIndex] = newRow;
+
+      if (JSON.stringify(originalRow) !== JSON.stringify(newRow)) {
+        moved = true;
+      }
+    });
+
+    if (moved) {
+      this.addTile();
+    }
+    this.checkGameStatus();
+  }
+
+  moveRight() {
+    if (this.status !== 'playing') {
+      return;
+    }
```

Comment:
why do you need this check in those functions if you have it here where you call these functions
<img width="592" alt="image" src="https://github.com/user-attachments/assets/d191f28f-9973-4bad-b0e2-2f6c8ce0cdd9">